src/datasources/accounts/accounts.datasource.ts

```diff
@@ -58,25 +94,47 @@ export class AccountsDatasource implements IAccountsDatasource {
     const { count } = await this
       .sql`DELETE FROM accounts WHERE address = ${address}`;
 
+    const { key } = CacheRouter.getAccountCacheDir(address);
+    await this.cacheService.deleteByKey(key);
```

Comment:
If this throws, the cache will remain whereas it will be removed from the database.

Reply:
Good catch, theoretically if the `DELETE` throws it won't be deleted, but maybe there are some scenarios where the deletion is performed and the cache invalidation is not, so I'm wrapping this on a `try-finally` statement in [cd76e37](https://github.com/safe-global/safe-client-gateway/pull/1760/commits/cd76e374714c6582c39d3842a0837cda1f4a54fa)